core/src/main/java/pt/webdetails/cda/utils/QueryParameters.java

```diff
@@ -104,4 +113,50 @@ private static Object getParam( List<String> paramValues ) {
     }
     return paramValues;
   }
+
+  /**
+   * Local implementation of the MultivaluedHashMap that is bundled on a javax ws rs api dependency.
+   * It's custom implemented here to avoid bringing a dependency just for the simple use of this.
+   * @param <K> The keys type.
+   * @param <V> The values type.
+   */
+  protected class MultiValueQueryParameter<K, V> extends HashMap<K, List<V>> implements MultivaluedMap<K, V> {
```

Comment:
why implement the interface if we dont need it? can't we just use a Map<String,List<String>>?

Reply:
Excelent point